bleak/__init__.py

```diff
@@ -204,6 +206,55 @@ def set_scanning_filter(self, **kwargs):
         )
         self._backend.set_scanning_filter(**kwargs)
 
+    def __aiter__(self) -> AsyncIterator[Tuple[BLEDevice, AdvertisementData]]:
+        self._backend.new_devices = asyncio.Queue()
+        return self
+
+    async def __anext__(self) -> Tuple[BLEDevice, AdvertisementData]:
+        return await self._backend.new_devices.get()
+
+    @overload
+    def devices(
+        self, timeout: Optional[float] = None, return_adv: Literal[False] = False
+    ) -> AsyncIterator[BLEDevice]:
+        ...
+
+    @overload
+    def devices(
+        self, timeout: Optional[float] = None, return_adv: Literal[True] = True
+    ) -> AsyncIterator[Tuple[BLEDevice, AdvertisementData]]:
+        ...
```

Comment:
Omitting `async` in typing overloads is deliberate (lost some time on this :facepalm:), see https://stackoverflow.com/questions/68905848/how-to-correctly-specify-type-hints-with-asyncgenerator-and-asynccontextmanager/68911014#68911014 and https://github.com/python/mypy/issues/5385#issuecomment-407281656